### PR TITLE
Template: Add Name List section of a Resource Fork

### PR DESCRIPTION
**Pull request by ahelsley**
1. Add Name List section of a Resource Fork.
2. Adds an explicit Resources section to more closely resemble https://developer.apple.com/library/archive/documentation/mac/pdf/MoreMacintoshToolbox.pdf (page 1-121 (around page 151 of the PDF)).

### CODE REVIEW

**Comment by kainjow** [comment]
I'm not sure how useful this list is. This data is the same that's already available in the Data > Reference Lists section.

File: templates/Apple/ResourceFork.tcl [hunk 1]
```diff
@@ -58,6 +72,16 @@ section "File Format" {
                 incr i
             }
         }
+
+        goto [expr {$map_offset + $name_list_offset}]
+        section "Name List" {
+            for {set i 0} {![end]} {incr i} {
+                section "Name $i" {
+                    set num_name_bytes [uint8 "Length"]
```
Comment (kainjow) [inline templates/Apple/ResourceFork.tcl:80]: are these utf8? or macroman?

**Comment by ahelsley** [comment]
> I'm not sure how useful this Resources list is. This data is the same that's already available in the Data > Reference Lists section, right?

The Name List section lets you actually read the names.  The Resources List entries only include the offset to the name in the Name List, not the name itself.  I have another change I'm interested in sending a PR for which builds on this to show the name in the Resource List.  Shall I bring that into the branch as a new commit so you can review the sequence of changes?

File: templates/Apple/ResourceFork.tcl (hunk 1 above)
Comment (ahelsley) [reply templates/Apple/ResourceFork.tcl:80]: Most likely Mac Roman.  Does HexFiend use the TCL encodings and their names (`encoding names` in tclsh)?  If so I can replace the `ascii` below with `str $num_name_bytes "macRoman" "Name"`.

Comment (ahelsley) [reply templates/Apple/ResourceFork.tcl:153]: I tested it out and switched from `ascii ...` to `str ... "macRoman" ...`.

**Comment by kainjow** [comment]
cool!

**Pull request merged**